variables/collection.go

```diff
@@ -42,9 +43,10 @@ type Collection struct {
 }
 
 // NewCollection creates a new Collection to be used in the context of a target.
-func NewCollection(console conslogging.ConsoleLogger, target domain.Target, platform specs.Platform, gitMeta *gitutil.GitMetadata, overridingVars *Scope, globalImports map[string]domain.ImportTrackerVal) *Collection {
+func NewCollection(console conslogging.ConsoleLogger, target domain.Target, platform specs.Platform, gitMeta *gitutil.GitMetadata,
+	defaultArgs DefaultArgs, overridingVars *Scope, globalImports map[string]domain.ImportTrackerVal, ftrs *features.Features) *Collection {
```

Comment:
This function signature is getting quite long.

Thoughts on budling them into a struct?

```golang
type NewCollectionParams struct {
    console conslogging.ConsoleLogger
    target    domain.Target
    // etc
}

func NewCollection(params NewCollectionParams) *Collection { 
    // ...
}

Reply:
Sounds good to me. We've used the term `opts` or `options` instead of `params` in other places in the code, but the idea is the same.